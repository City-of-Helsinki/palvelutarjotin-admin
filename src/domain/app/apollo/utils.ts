import apolloClient from './apolloClient';

export const clearApolloCache = async () => {
  apolloClient.stop();
  await Promise.all([apolloClient.resetStore(), apolloClient.clearStore()]);
};
