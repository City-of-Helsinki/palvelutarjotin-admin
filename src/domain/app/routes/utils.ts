import { ROUTES } from './constants';

export const getCmsPath = (slug?: string | null) => {
  return ROUTES.CMS_PAGE.replace('/:slug', slug ?? '');
};
