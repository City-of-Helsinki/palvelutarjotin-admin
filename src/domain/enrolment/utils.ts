import { EnrolmentFieldsFragment } from '../../generated/graphql';

export const getEnrolmentFields = (
  enrolment: EnrolmentFieldsFragment | undefined | null
) =>
  enrolment
    ? {
        extraNeeds: enrolment.studyGroup.extraNeeds,
        language:
          enrolment.person?.language || enrolment.studyGroup.person.language,
        personInfo: [
          enrolment.person?.phoneNumber,
          enrolment.person?.emailAddress,
        ].filter((e) => e),
        studyGroupPersonInfo: [
          enrolment.studyGroup.person.phoneNumber,
          enrolment.studyGroup.person.emailAddress,
        ].filter((e) => e),
      }
    : {};
