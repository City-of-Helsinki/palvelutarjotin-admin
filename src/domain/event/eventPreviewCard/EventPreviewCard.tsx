/* eslint-disable jsx-a11y/anchor-is-valid */
import { IconLocation } from 'hds-react';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';

import { EventFieldsFragment } from '../../../generated/graphql';
import useLocale from '../../../hooks/useLocale';
import IconClock from '../../../icons/IconClock';
import PlaceText from '../../place/placeText/PlaceText';
import EventKeywords from '../eventKeywords/EventKeywords';
import {
  getEventFields,
  getEventPlaceholderImage,
  getNextOccurrenceDateStr,
} from '../utils';
import styles from './eventPreviewCard.module.scss';

interface Props {
  event: EventFieldsFragment;
  link: string;
}

const EventPreviewCard: React.FC<Props> = ({ event, link }) => {
  const { t } = useTranslation();
  const locale = useLocale();

  const { id, eventName, shortDescription, imageUrl } = getEventFields(
    event,
    locale
  );
  const time = getNextOccurrenceDateStr(event, locale, t);

  return (
    <Link to={link} className={styles.eventCard}>
      <div
        className={styles.imageWrapper}
        style={{
          backgroundImage: `url(${
            imageUrl || getEventPlaceholderImage(id || '')
          })`,
        }}
      ></div>
      <div className={styles.contentWrapper}>
        <div className={styles.titleWrapper}>
          <div className={styles.title}>{eventName}</div>
          <div className={styles.description}>{shortDescription}</div>
        </div>
        <div className={styles.occurrenceInfoWrapper}>
          <div className={styles.textWithIcon}>
            <IconClock />
            {time}
          </div>
          <div className={styles.textWithIcon}>
            <IconLocation />
            <PlaceText placeId={event.location?.id || ''} />
          </div>
        </div>
        <EventKeywords event={event} />
      </div>
    </Link>
  );
};

export default EventPreviewCard;
