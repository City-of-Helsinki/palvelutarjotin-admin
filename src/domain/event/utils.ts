import { SUPPORT_LANGUAGES } from '../../constants';
import { EventQuery } from '../../generated/graphql';
import { Language } from '../../types';
import { EVENT_PLACEHOLDER_IMAGES } from './constants';
/**
 * Get event placeholder image url
 * @param {string} id
 * @return {string}
 */
export const getEventPlaceholderImage = (id: string): string => {
  const numbers = id.match(/\d+/g);
  const sum = numbers
    ? numbers.reduce((prev: number, cur: string) => prev + Number(cur), 0)
    : 0;
  const index = sum % 4;

  return EVENT_PLACEHOLDER_IMAGES[index];
};

/**
 * Get first availanle language of an event
 * @param {object} eventData
 * @return {string}
 */
export const getFirstAvailableLanguage = (eventData: EventQuery): Language => {
  if (eventData.event?.name.en) return SUPPORT_LANGUAGES.EN;
  if (eventData.event?.name.fi) return SUPPORT_LANGUAGES.FI;
  if (eventData.event?.name.sv) return SUPPORT_LANGUAGES.SV;

  return SUPPORT_LANGUAGES.FI;
};
