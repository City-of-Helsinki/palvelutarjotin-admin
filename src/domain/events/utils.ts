import { isFuture } from 'date-fns';
import forEach from 'lodash/forEach';

import { EventFieldsFragment } from '../../generated/graphql';

export const hasOccurrences = (event: EventFieldsFragment) => {
  return Boolean(event.pEvent?.occurrences.edges.length);
};

export const hasComingOccurrences = (event: EventFieldsFragment) => {
  let hasComingItems = false;

  forEach(event.pEvent?.occurrences.edges, (edge) => {
    if (edge?.node?.startTime && isFuture(new Date(edge?.node?.startTime))) {
      hasComingItems = true;
      return false;
    }
  });

  return Boolean(hasComingItems);
};
