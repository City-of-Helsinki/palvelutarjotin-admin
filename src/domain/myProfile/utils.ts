import {
  OrganisationNodeFieldsFragment,
  PersonFieldsFragment,
} from '../../generated/graphql';
import { MyProfileFormFields } from './myProfileForm/MyProfileForm';

export const getMyProfilePayload = (values: MyProfileFormFields) => ({
  name: values.name,
  phoneNumber: values.phoneNumber,
  emailAddress: values.emailAddress,
  organisations: values.organisations,
});

export const getSelectedOrganisation = (
  myProfile: PersonFieldsFragment,
  activeOrganisation: string | null
): OrganisationNodeFieldsFragment | null => {
  const organisations =
    myProfile.organisations.edges.map((edge) => ({
      ...(edge?.node as OrganisationNodeFieldsFragment),
    })) || [];

  const organisation = activeOrganisation
    ? organisations.find((item) => item.id === activeOrganisation)
    : null;

  return organisation || organisations[0];
};
