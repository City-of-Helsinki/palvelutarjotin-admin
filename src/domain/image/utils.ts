import { ImageDocument, ImageQuery } from '../../generated/graphql';
import { initializeApolloClient } from '../app/apollo/apolloClient';

export const getImageName = (id: string) => {
  const apolloClient = initializeApolloClient();
  const data = apolloClient.readQuery<ImageQuery>({
    query: ImageDocument,
    variables: { id },
  });
  return data?.image?.name;
};
