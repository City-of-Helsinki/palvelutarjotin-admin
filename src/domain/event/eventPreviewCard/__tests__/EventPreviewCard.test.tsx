import * as React from 'react';

import {
  fakeEvent,
  fakeKeyword,
  fakeLocalizedObject,
  fakeOccurrences,
  fakePEvent,
} from '../../../../utils/mockDataUtils';
import { render, screen } from '../../../../utils/testUtils';
import EventPreviewCard from '../EventPreviewCard';

it('renders event information correctly and matches snapshot', () => {
  vi.setSystemTime(new Date(2021, 9, 20));
  const eventName = 'Testinimi';
  const description = 'Testikuvaus';
  const keywords = ['avainsana1', 'avainsana2'];
  const href = '/test-link';
  const event = fakeEvent({
    name: fakeLocalizedObject(eventName),
    shortDescription: fakeLocalizedObject(description),
    keywords: keywords.map((keyword) =>
      fakeKeyword({ name: fakeLocalizedObject(keyword) })
    ),
    pEvent: fakePEvent({
      occurrences: fakeOccurrences(2, [
        {
          startTime: '2021-09-10T13:17:09Z',
          endTime: '2021-09-10T13:17:09Z',
        },
        // this is the next occurrence in the future (should be rendered)
        {
          startTime: '2021-10-22T13:17:09Z',
          endTime: '2021-10-22T14:17:09Z',
        },
      ]),
    }),
  });

  const { container } = render(<EventPreviewCard event={event} link={href} />);
  const link = screen.getByRole('link', {
    name: /testinimi/i,
  });
  expect(link).toHaveAttribute('href', href);
  screen.getByText(eventName);
  screen.getByText(description);
  screen.getByText('Maksuton');
  keywords.forEach((k) => screen.getByText(k));

  // render next occurrence time
  screen.getByText('perjantaina 22.10 klo 16:17');
  expect(container).toMatchSnapshot();
});

it('render multiday occurrence dates correctly', () => {
  vi.setSystemTime(new Date(2021, 9, 20));

  render(
    <EventPreviewCard
      event={fakeEvent({
        pEvent: fakePEvent({
          occurrences: fakeOccurrences(2, [
            {
              startTime: '2021-09-10T13:17:09Z',
              endTime: '2021-09-10T13:17:09Z',
            },
            // this is the next occurrence in the future (should be rendered)
            {
              startTime: '2021-10-22T13:17:09Z',
              endTime: '2021-10-28T14:17:09Z',
            },
          ]),
        }),
      })}
      link="/"
    />
  );
  screen.getByText('22.10.2021 â€“ 28.10.2021');
});
