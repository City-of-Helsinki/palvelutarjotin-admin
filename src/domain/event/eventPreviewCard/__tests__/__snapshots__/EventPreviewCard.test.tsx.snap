// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders event information correctly and matches snapshot 1`] = `
<div>
  <a
    class="eventCard"
    href="/test-link"
  >
    <div
      class="imageWrapper"
      style="background-image: url(https://api.hel.fi/linkedevents-test/media/images/test.png);"
    />
    <div
      class="contentWrapper"
    >
      <div
        class="titleWrapper"
      >
        <div
          class="title"
        >
          Testinimi
        </div>
        <div
          class="description"
        >
          Testikuvaus
        </div>
      </div>
      <div
        class="occurrenceInfoWrapper"
      >
        <div
          class="textWithIcon"
        >
          <svg
            class=""
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill-rule="evenodd"
              stroke-width="1"
            >
              <path
                d="M11.80416,-2.30926389e-13 C18.32352,-2.30926389e-13 23.60928,5.2848 23.60928,11.80416 C23.60928,18.32352 18.32352,23.60928 11.80416,23.60928 C5.2848,23.60928 -3.37507799e-14,18.32352 -3.37507799e-14,11.80416 C-3.37507799e-14,5.2848 5.2848,-2.30926389e-13 11.80416,-2.30926389e-13 Z M11.80416,2.5296 C6.6912,2.5296 2.53056,6.69024 2.53056,11.80416 C2.53056,16.91904 6.6912,21.07968 11.80416,21.07968 C16.91808,21.07968 21.07968,16.91904 21.07968,11.80416 C21.07968,6.69024 16.91808,2.5296 11.80416,2.5296 Z M13.068768,5.842464 L13.068768,11.280864 L16.665888,14.877024 L14.878368,16.665504 L10.539168,12.328224 L10.539168,5.842464 L13.068768,5.842464 Z"
              />
            </g>
          </svg>
          perjantaina 22.10 klo 16:17
        </div>
        <div
          class="textWithIcon"
        >
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M11.967 1.5c2.06 0 4.12.778 5.69 2.334 3.143 3.111 2.93 7.96 0 11.268l-.622.709c-2.612 2.991-4.066 4.96-5.068 6.937-1.073-2.13-2.682-4.249-5.689-7.646-2.93-3.308-3.143-8.157 0-11.268A8.06 8.06 0 0111.967 1.5zm.032 2a6.072 6.072 0 00-4.3 1.762A5.606 5.606 0 006.002 9.41c.02 1.573.648 3.134 1.766 4.398l.66.752c1.59 1.823 2.717 3.239 3.573 4.503.975-1.437 2.292-3.063 4.233-5.255 1.118-1.264 1.746-2.825 1.766-4.398a5.616 5.616 0 00-1.698-4.15A6.077 6.077 0 0011.999 3.5zM12 6a3.5 3.5 0 110 6.999A3.5 3.5 0 0112 6zm0 2c-.827 0-1.5.673-1.5 1.5S11.173 11 12 11s1.5-.673 1.5-1.5S12.827 8 12 8z"
                fill="currentColor"
              />
            </g>
          </svg>
        </div>
      </div>
      <div
        class="eventKeywords"
      >
        <div
          class="keyword"
        >
          Maksuton
        </div>
        <div
          class="keyword"
        >
          avainsana1
        </div>
        <div
          class="keyword"
        >
          avainsana2
        </div>
      </div>
    </div>
  </a>
</div>
`;
